cpp/src/arrow/result_test.cc

```diff
@@ -313,7 +313,7 @@ TYPED_TEST(ResultTest, MoveConstructorNonOkStatus) {
   Result<typename TypeParam::value_type> result2(std::move(result1));
 
   // Verify that the status of the donor object was updated.
-  EXPECT_FALSE(result1.ok());
+  //   EXPECT_FALSE(result1.ok());
```

Comment:
Testing leftover for this file?

Reply:
Basically, the condition that was being tested is not true anymore. I'll remove the corresponding test lines.